lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/services/LibertyProjectsManager.java

```diff
@@ -45,8 +47,63 @@ private LibertyProjectsManager() {
 
     public void setWorkspaceFolders(List<WorkspaceFolder> workspaceFolders) {
         for (WorkspaceFolder folder : workspaceFolders) {
-            LibertyWorkspace libertyWorkspace = new LibertyWorkspace(folder.getUri());
-            this.libertyWorkspaceFolders.add(libertyWorkspace);
+            // Add logic here to see if child folders are sub-modules in a multi-module project.
+            // If child folder is a Liberty project (has src/main/liberty/config/server.xml),
+            // then create a LibertyWorkspace for it and do not create one for the parent folder.
+
+            String workspaceUriString = folder.getUri();
+            URI workspaceUri =  URI.create(workspaceUriString);
+            Path workspacePath = Paths.get(workspaceUri);
+            Path serverXmlPath = Paths.get("src", "main", "liberty", "config", "server.xml");
+            List<Path> serverXmlFiles = null;
+
+            try {
+                serverXmlFiles = LibertyUtils.findFilesInDirectory(workspacePath, serverXmlPath);
+            } catch (IOException e) {
+                LOGGER.warning("Received exception while searching for server.xml files in: " + workspacePath + " exception: " + e.getMessage());
+            }
+
+            if ((serverXmlFiles == null) || serverXmlFiles.isEmpty() || (serverXmlFiles.size() == 1)) {
+                LOGGER.info("Adding Liberty workspace: " + workspaceUriString);
+                LibertyWorkspace libertyWorkspace = new LibertyWorkspace(workspaceUriString);
+                this.libertyWorkspaceFolders.add(libertyWorkspace);
+            } else {
+                boolean addedSubModule = false;
+
+                List<Path> childrenDirs = null;
+                String lastChildDirPath = null;
+
+                try {
+                    childrenDirs = Files.walk(workspacePath, 1)
+                                        .filter(Files::isDirectory)
+                                        .collect(Collectors.toList());
+
+                    for (Path nextChildDir : childrenDirs) {
+                        lastChildDirPath = nextChildDir.toUri().toString().replace("///", "/");;
+                        if (nextChildDir.equals(workspacePath)) {
+                            continue;
+                        }
+                        List<Path> serverXmlFile = LibertyUtils.findFilesInDirectory(nextChildDir, serverXmlPath);
+                        if (!serverXmlFile.isEmpty()) {
```

Comment:
Maybe we don't need the Maven check.  It sounds like this logic doesn't really intersect with the reactor or aggregate aspects of the multi-module build.  If like we were saying last week, the project scope is mainly about drawing a boundary around server config includes... that would seem to stop at the submodule level.. and it also doesn't seem to require anything in LMP not in LGP.